File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-2.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        seen = set()
        length = 0
        l, r = 0, 0
        while r <= len(s) - 1:
            if not s[r] in seen:
                seen.add(s[r])
                length = max(length, r-l+1)
                r+=1
            elif s[r] in seen:
                while s[r] in seen:
                    seen.remove(s[l])
                    l+=1
        return length
            

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-2.js
class Solution {
    /**
     * @param {number[]} prices
     * @return {number}
     */
    maxProfit(prices) {
        let l = 0;
        let profit = 0;
        for (let r = 0; r < prices.length; r++) { 
            if (prices[r] > prices[l]) { 
                profit = Math.max(profit, (prices[r] - prices[l]));
            } else { 
                l = r;
            }
        }
        return profit;
    }
}
